### PR TITLE
A light improving of the component search-all-results.component.ts

### PR DESCRIPTION
**Pull request by sofiiamatsyhin**
[1162](https://github.com/ita-social-projects/GreenCity/issues/1162)

### CODE REVIEW

**Review by yulkats — APPROVED** [review]

File: src/app/component/layout/components/search-all-results/search-all-results.component.ts [hunk 1]
```diff
@@ -48,13 +48,21 @@ export class SearchAllResultsComponent implements OnInit, OnDestroy {
 
   private getSearchData(data: SearchModel): void {
     this.getNews(data.ecoNews);
-    // tslint:disable-next-line:no-unused-expression
-    data.countOfResults ? this.itemsFound = data.countOfResults : null;
+    if (data.countOfResults) {
+      this.itemsFound = data.countOfResults;
+    } else {
+      this.itemsFound = null;
+    }
```
Comment (mplesha) [inline src/app/component/layout/components/search-all-results/search-all-results.component.ts:55]: using ternary was good, but wrong styled. should be rewrite ternary only, not change it on `if`:
```js
this.itemsFound = ...
```

File: src/app/component/layout/components/search-all-results/search-all-results.component.ts [hunk 2]
```diff
@@ -48,13 +48,21 @@ export class SearchAllResultsComponent implements OnInit, OnDestroy {
...
   }
 
   private getNews(news): void {
-    (news && news.length > 0) ? (this.isSearchFound = true, this.elements = news) : (this.isSearchFound = false);
+    if (news && news.length > 0) {
+      this.isSearchFound = true;
+      this.elements = news;
+    } else {
+      this.isSearchFound = false;
+    }
```
Comment (mplesha) [inline src/app/component/layout/components/search-all-results/search-all-results.component.ts:65]: the same as previous. But a slightly different, I'd propose:
```js
this.isSearchFound = news && news.length;
this.elements = this.isSearchFound ? news : this.elements;
```
- looks more simplify as both

Comment (sofiiamatsyhin) [reply src/app/component/layout/components/search-all-results/search-all-results.component.ts]: Thanks, fixed

File: src/app/component/layout/components/search-all-results/search-all-results.component.ts (hunk 1 above)
Comment (sofiiamatsyhin) [reply src/app/component/layout/components/search-all-results/search-all-results.component.ts]: Fixed

**Comment by mplesha** [comment]
To fully incorporate all the improvements in this component should rewrite those:
```ts
if (event[VALUE].length > 0) {
      this.inputValue = event[VALUE];
      this.searchSubscription = this.search.getAllSearch(this.inputValue, this.inputValues[0])
        .subscribe(this.getSearchData.bind(this));
```
-- unnecessary `bind`
and
```ts
for (let i = 0; i < 9; i++) {
      if (splicedData[i]) {
        this.displayedElements.push(splicedData[i]);
      }
    }
```

File: src/app/component/layout/components/search-all-results/search-all-results.component.ts [hunk 3]
```diff
@@ -40,35 +40,30 @@ export class SearchAllResultsComponent implements OnInit, OnDestroy {
...
   }
 
   private spliceResults(): void {
     const splicedData = this.elements.splice(0, 9);
-    for (let i = 0; i < 9; i++) {
-      if (splicedData[i]) {
-        this.displayedElements.push(splicedData[i]);
-      }
-    }
+    this.displayedElements = [...splicedData.map(elem => elem)];
```
Comment (mplesha) [inline src/app/component/layout/components/search-all-results/search-all-results.component.ts:62]: Firstly - `map` returns new instance of array, so - you shouldn't use `spread` operations `...`, it's unneeded . Just
```js
this.displayedElements = splicedData.map(elem => elem)
```
Secondly - your changes don't quite correspond to the original code. There will be pushed to the array only if element of `spliceData` is true.
So, change the `map` method in your code on the `filter`.

Comment (sofiiamatsyhin) [reply src/app/component/layout/components/search-all-results/search-all-results.component.ts]: Fixed

**Pull request merged**